mars/web/session.py

```diff
@@ -167,20 +168,36 @@ def fetch(self, *tileables, **kw):
 
         results = list()
         for tileable in tileables:
-            key = tileable.key
+            if tileable.key not in self._executed_tileables and isinstance(tileable.op, TensorIndex):
+                key = tileable.inputs[0].key
+                indexes = tileable.op.indexes
+                if not all(isinstance(ind, slice) for ind in indexes):
+                    raise ValueError('Only support fetch data slices')
+            else:
+                key = tileable.key
+                indexes = []
 
             if key not in self._executed_tileables:
                 raise ValueError('Cannot fetch the unexecuted tileable')
 
+            slice_str = '-'.join(['{0},{1},{2}'.format(s.start, s.stop, s.step) for s in indexes])
```

Comment:
Better to add a util to do this, so that if we want to modify the protocol, we can find the implementation quickly.